src/moonlink/src/storage/filesystem/accessor/filesystem_accessor.rs

```diff
@@ -80,17 +82,32 @@ impl FileSystemAccessor {
                         bucket,
                         endpoint,
                         disable_auth,
+                        cred_path,
                         ..
                     } => {
-                        let mut builder = services::Gcs::default()
-                            .root("/")
-                            .bucket(bucket)
-                            .endpoint(endpoint);
+                        // Check invariants: when authentication disabled, no need to provide credential path.
+                        if *disable_auth {
+                            assert!(cred_path.is_none());
+                        }
+
+                        let mut builder = services::Gcs::default().root("/").bucket(bucket);
+                        if let Some(endpoint) = endpoint {
+                            builder = builder.endpoint(endpoint);
+                        }
+
+                        // Set up authentication.
                         if *disable_auth {
                             builder = builder
                                 .disable_config_load()
                                 .disable_vm_metadata()
                                 .allow_anonymous();
+                        } else {
+                            let resolved_cred_path = gcs_cred_utils::get_credential_path(cred_path);
+                            // Failed credential resolution by default falls back to VM metadata server, explicitly disable.
```

Comment:
This is opendal implementation bug / feature, it automatically falls back to metadata server if credential access failure; but these are two completely different authentication.